File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/SharedPreferenceManager.java
package com.dani_chuks.andeladeveloper.starwars.data;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;


public class SharedPreferenceManager {

    private final SharedPreferences sharedPreferences;
    private final Context context;

    public SharedPreferenceManager(@NonNull final SharedPreferences sharedPreferences, @NonNull final Context context) {
        this.sharedPreferences = sharedPreferences;
        this.context = context;
    }


    public void setDataTypeFetchedOnce(String resourceName){
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(resourceName, true);
        editor.apply();
    }

    public boolean isDataTypeFetchedOnce(String resourceName){
        return sharedPreferences.getBoolean(resourceName, false);
    }

    public void setResourceNextPage(String resourceName, int nextPage){
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt(resourceName, nextPage);
        editor.apply();
    }

    public int getResourceNextPage(String resourceName){
        return sharedPreferences.getInt(resourceName, 1);
    }
}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/SharedPreferenceManager.kt
package com.dani_chuks.andeladeveloper.starwars.data

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit
import javax.inject.Inject


class SharedPreferenceManager @Inject constructor(private val sharedPreferences: SharedPreferences, private val context: Context) {

    fun setDataTypeFetchedOnce(resourceName: String) {
        sharedPreferences.edit { putBoolean(resourceName, true) }
    }

    fun isDataTypeFetchedOnce(resourceName: String): Boolean =
            sharedPreferences.getBoolean(resourceName, false)


    fun setResourceNextPage(resourceName: String, nextPage: Int) {
        sharedPreferences.edit { putInt(resourceName, nextPage) }
    }

    fun getResourceNextPage(resourceName: String): Int =
            sharedPreferences.getInt(resourceName, 1)
}
